macaroon_service.go

```diff
@@ -231,12 +233,54 @@ func (ms *MacaroonService) Start() error {
 
 	// There are situations in which we don't want a macaroon to be created
 	// on disk (for example when running inside LiT stateless integrated
-	// mode). For any other cases, we create macaroon files in the default
-	// directory.
-	if ms.cfg.StatelessInit || lnrpc.FileExists(ms.cfg.MacaroonPath) {
+	// mode).
+	if ms.cfg.StatelessInit {
 		return nil
 	}
 
+	// If there are no required permissions, then we can't create a macaroon
+	// and we will return an error.
+	if len(ms.cfg.RequiredPerms) == 0 {
+		return fmt.Errorf("the config field RequiredPerms must to be " +
+			"set and must contain elements to create a macaroon")
```

Comment:
I think Oli was saying we should move this check into `NewMacaroonService` right?

Reply:
Oh sorry, missed that part. Will fix this!